slack_sdk/web/client.py

```diff
@@ -4111,6 +4111,20 @@ def files_completeUploadExternal(
         )
         if channels:
             kwargs["channels"] = ",".join(channels)
+        
+        # Parse Block objects and serialize blocks/attachments to JSON
+        _parse_web_class_objects(kwargs)
+        
+        # Serialize blocks to JSON if present and not already a string
+        blocks = kwargs.get("blocks")
+        if blocks is not None and not isinstance(blocks, str):
+            kwargs["blocks"] = json.dumps(blocks)
+        
+        # Serialize attachments to JSON if present and not already a string
+        attachments = kwargs.get("attachments")
+        if attachments is not None and not isinstance(attachments, str):
+            kwargs["attachments"] = json.dumps(attachments)
+        
         return self.api_call("files.completeUploadExternal", params=kwargs)
```

Comment:
```suggestion
        _parse_web_class_objects(kwargs)
        kwargs = _remove_none_values(kwargs)
        return self.api_call("files.completeUploadExternal", json=kwargs)
```

👁️‍🗨️ suggestion: We can perhaps prefer to use JSON for these nested values like this!